### PR TITLE
Set translatorsDir to %CurProcD%/../module/zotero/Translators by default

### PR DESCRIPTION
**Pull request by kba**
Since it's recommended to clone the submodules (including translators) recursively, `modules/zotero/translators` will be the right source for the translators in most cases. If users don't configure anything, this should be the default location to look for, not the hard-coded location.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -89,4 +90,4 @@ echo "resource zotero translation-server/resource/" >> "$BUILDDIR/chrome.manifes
 
 find "$BUILDDIR" -depth -type d -name .svn -exec rm -rf {} \;
 find "$BUILDDIR" -name .DS_Store -exec rm -rf \;
-find "$BUILDDIR" -name '._*' -exec rm -rf \;
\ No newline at end of file
+find "$BUILDDIR" -name '._*' -exec rm -rf \;
```
Comment (dstillman) [inline build.sh]: Undo

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,2 @@
 build
-xulrunner-sdk
+xulrunner*
```
Comment (dstillman) [inline .gitignore]: Undo: unrelated, the gzips from Mozilla use xulrunner-sdk, and the build script has xulrunner-sdk hardcoded

File: build.sh [hunk 3]
```diff
@@ -70,6 +70,7 @@ mkdir "$BUILDDIR/translation-server/resource"
 cp -r "$EXTENSIONDIR/resource/q.js" "$BUILDDIR/translation-server/resource"
 
 cp "$CWD/config.js" "$BUILDDIR/defaults/pref"
+sed -i "s,/Users/simon/Desktop/Development/FS/zotero,$EXTENSIONDIR," "$BUILDDIR/defaults/pref/config.js"
```
Comment (dstillman) [inline build.sh]: If we're going to change this, we should get rid of Simon's local path in the config file. I think it'd be better to have the code (in server_translation.js) automatically use `modules/zotero/translators` if `translation-server.translatorsDirectory` is empty and otherwise use the provided path. Then this doesn't need to be changed in the build script at all, and moving the root directory after building won't break the install.

Comment (dstillman) [inline build.sh]: For getting the root path, I'm not sure if the directory service is available in XULRunner, but https://developer.mozilla.org/en-US/Add-ons/Code_snippets/File_I_O#Getting_files_in_special_directories may be helpful.

File: .gitignore (hunk 2 above)
Comment (kba) [inline .gitignore]: This is to prevent adding a downloaded tarball, like `xulrunner-41.0.en-US.linux-x86_64.sdk.tar.bz2`.

Comment (dstillman) [inline .gitignore]: That file can be deleted after extraction.

File: build.sh (hunk 3 above)
Comment (kba) [inline build.sh]: Seems reasonable, I'll look into it.

File: build.sh (hunk 1 above)
Comment (kba) [inline build.sh]: I'd have to investigate how to prevent my editor from adding a trailing newline but I can. Any particular reason against trailing newline?

Comment (dstillman) [inline build.sh]: Just as a general rule, there shouldn't be unrelated lines in commits. You can use `git checkout -p` to revert individual lines in the diff before committing.

Comment (zuphilip) [inline build.sh]: Maybe one can also add the missing newlines for all files in a separate commit beforehand?

Comment (dstillman) [inline build.sh]: We don't need to do that. Unrelated lines should just be removed from commits — it's standard practice.

**Comment by kba** [comment]
I've updated it with code that defaults to [`%CurProcD%`](https://developer.mozilla.org/en-US/Add-ons/Code_snippets/File_I_O)`/../modules/zotero/translators` if the `translatorsDirectory` pref is falsy.

I'm not sure if `CurProcD` and navigating up is the best choice but it's the only special directory I got to work.

Do you think this is the right approach? I'll happily reformat and rebase the pull request if so.

File: src/server_translation.js [hunk 4]
```diff
@@ -66,18 +66,27 @@ Zotero.Server.Translation = new function() {
...
+	this.translatorsDirPath = null;
 	
 	/**
 	 * Initializes translation server by reading files from local translators directory
 	 */
 	this.init = function() {
+		// If "translatorsDirectory" pref is empty, default 
+		// to %CurProcD%/../modules/zotero/translators
+		this.translatorsDirPath = Zotero.Prefs.get("translatorsDirectory") ||
+			Components.classes["@mozilla.org/file/directory_service;1"].
```
Comment (dstillman) [inline src/server_translation.js]: Just for our style, move periods to beginnings of lines

File: src/server_translation.js [hunk 5]
```diff
@@ -462,8 +471,8 @@ Zotero.Server.Translation.Refresh.prototype = {
 			createInstance(Components.interfaces.nsIProcess);
 		proc.init(bash);
 		
-		var translatorsDir = "'"+Zotero.Prefs.get("translatorsDirectory").replace("'", "'\\''", "g")+"'";
-		var args = ["-c", "cd "+translatorsDir+" && git pull origin master"];
+		var translatorsDirPathClean = "'"+Zotero.Server.Translation.translatorsDirPath.replace("'", "'\\''", "g")+"'";
+		var args = ["-c", "cd "+translatorsDirPathClean+" && pwd && git pull origin master"];
```
Comment (dstillman) [inline src/server_translation.js]: The `pwd` isn't necessary, right?

**Comment by dstillman** [comment]
Looks good otherwise. Thanks!

File: src/server_translation.js (hunk 5 above)
Comment (kba) [inline src/server_translation.js]: Right, debug artifact. I'll remove it.

File: src/server_translation.js (hunk 4 above)
Comment (kba) [inline src/server_translation.js]: I tried to be consistent with the style in that file, e.g. https://github.com/zotero/translation-server/blob/master/src/server_translation.js#L457 and https://github.com/zotero/translation-server/blob/master/src/server_translation.js#L461.
